### PR TITLE
Use the smart_proxy_plugin.yml reusable GHA workflow

### PR DESCRIPTION
**Pull request by ekohl**
This is largely the same as test-gem, except it verifies that it at least implements support for Ruby versions that the Smart Proxy wants. It also implements the version selector by respecting SMART_PROXY_BRANCH in the Gemfile.

It also cleans up the Jenkins integration.

### CODE REVIEW

**Comment by evgeni** [comment]
Correct, it's a bad test that was (accidentally) broken by https://github.com/theforeman/smart_proxy_dynflow/commit/d6e6a73e151f1a0c664569555ef78f922735ef01

File: test/unit/salt_runner_test.rb [hunk 1]
```diff
@@ -32,9 +32,13 @@ def test_capture_jid
...
         assert_nil runner.jid
-        assert_equal 1, runner.publish_exit_status(0)
+        runner.publish_exit_status(0)
+        updates = runner.generate_updates
+        assert_equal 1, updates[{:suspended_action => nil}].exit_status
         runner.publish_data('jid: 12345', 'stdout')
-        assert_equal 0, runner.publish_exit_status(0)
+        runner.publish_exit_status(0)
+        updates = runner.generate_updates
+        assert_equal 0, updates[{:suspended_action => nil}].exit_status
```
Comment (evgeni) [inline test/unit/salt_runner_test.rb:41]: @adamruzicka would you mind having a look here whether the test fix looks sane to you? :)

Comment (adamruzicka) [reply test/unit/salt_runner_test.rb:41]: Looks good, thank you.

**Review by adamruzicka — APPROVED** [review]

**Review by evgeni — APPROVED** [review]
formal ack of changes done by @ekohl

**Comment by ekohl** [comment]
Thanks for the test fixes

**Pull request merged**